services/chaospod.go

```diff
@@ -0,0 +1,637 @@
+								FieldPath: "status.podIP",
+							},
+						},
+					},
+					{
+						Name: env.InjectorPodName,
+						ValueFrom: &corev1.EnvVarSource{
+							FieldRef: &corev1.ObjectFieldSelector{
+								FieldPath: "metadata.name",
+							},
+						},
+					},
+					{
+						Name:  env.InjectorMountHost,
+						Value: "/mnt/host/",
+					},
+					{
+						Name:  env.InjectorMountProc,
+						Value: "/mnt/host/proc/",
+					},
+					{
+						Name:  env.InjectorMountSysrq,
+						Value: "/mnt/sysrq",
+					},
+					{
+						Name:  env.InjectorMountSysrqTrigger,
+						Value: "/mnt/sysrq-trigger",
+					},
+					{
+						Name:  env.InjectorMountCgroup,
+						Value: "/mnt/cgroup/",
+					},
+				},
+				VolumeMounts: []corev1.VolumeMount{ // define volume mounts required for disruptions to work
+					{
+						Name:      "run",
+						MountPath: "/run",
+					},
+					{
+						Name:      "sysrq",
+						MountPath: "/mnt/sysrq",
+					},
+					{
+						Name:      "sysrq-trigger",
+						MountPath: "/mnt/sysrq-trigger",
+					},
+					{
+						Name:      "cgroup",
+						MountPath: "/mnt/cgroup",
+					},
+					{
+						Name:      "host",
+						MountPath: "/mnt/host",
+						ReadOnly:  true,
+					},
+				},
+			},
+		},
+		Volumes: []corev1.Volume{ // declare volumes required for disruptions to work
+			{
+				Name: "run",
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: "/run",
+						Type: &hostPathDirectory,
+					},
+				},
+			},
+			{
+				Name: "proc",
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: "/proc",
+						Type: &hostPathDirectory,
+					},
+				},
+			},
+			{
+				Name: "sysrq",
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: "/proc/sys/kernel/sysrq",
+						Type: &hostPathFile,
+					},
+				},
+			},
+			{
+				Name: "sysrq-trigger",
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: "/proc/sysrq-trigger",
+						Type: &hostPathFile,
+					},
+				},
+			},
+			{
+				Name: "cgroup",
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: "/sys/fs/cgroup",
+						Type: &hostPathDirectory,
+					},
+				},
+			},
+			{
+				Name: "host",
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: "/",
+						Type: &hostPathDirectory,
+					},
+				},
+			},
+		},
+	}
+
+	if m.config.ImagePullSecrets != "" {
+		podSpec.ImagePullSecrets = []corev1.LocalObjectReference{
+			{
+				Name: m.config.ImagePullSecrets,
+			},
+		}
+	}
+
+	return podSpec
+}
+
+func (m *chaosPodService) removeFinalizerForChaosPod(chaosPod *corev1.Pod) error {
+	controllerutil.RemoveFinalizer(chaosPod, chaostypes.ChaosPodFinalizer)
+
+	if err := m.config.Client.Update(context.Background(), chaosPod); err != nil {
+		if strings.Contains(err.Error(), "latest version and try again") {
```

Comment:
why not call the function for this check from chaosv1beta1? IsUpdateConflictError

Reply:
I was not aware of this function 🙈

Reply:
Thanks 👍